documentation/book/con-resource-kafka.adoc

```diff
@@ -0,0 +1,803 @@
+----
+
+When using external ConfigMap remember to place your custom ConfigMap under `log4j.properties` key.
+
+The difference between these two options is that the latter is not validated and does not support default values.
+That means the user can supply any logging configuration, even if it is incorrect.
+The first option supports default values.
+
+
+[id='resources_json_config-{context}']
+== Resource limits and requests
+
+It is possible to configure {ProductPlatformName} resource limits and requests on for the `kafka`, `zookeeper` and `topicOperator` objects in the `Kafka` resource and for the `spec` object of the `KafkaConnect` resource.
+The object may have a `requests` and a `limits` property, each having the same schema, consisting of `cpu` and `memory` properties.
+The {ProductPlatformName} syntax is used for the values of `cpu` and `memory`.
+
+.Example fragment of a `Kafka` resource configuring resource limits and requests for the Kafka pods
+[source,json]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    resources:
+      requests:
+        cpu: "1"
+        memory: "2Gi"
+      limits:
+        cpu: "1"
+        memory: "2Gi"
+    # ...
+----
+
+:k8s-docs-version: v1-7
+:k8s-resource-request-limit-docs-link: https://{k8s-docs-version}.docs.kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
+
+`requests.memory`::
+The memory request for the container, corresponding directly to {k8s-resource-request-limit-docs-link}[`spec.containers[\].resources.requests.memory`] setting.
+{ProductPlatformName} will ensure the containers have at least this much memory by running the pod on a node with at
+least as much free memory as all the containers require. The field is optional with no default value.
+
+`requests.cpu`::
+The CPU request for the container, corresponding directly to {k8s-resource-request-limit-docs-link}[`spec.containers[\].resources.requests.cpu`] setting.
+{ProductPlatformName} will ensure the containers have at least the specified CPU by running the pod on a node with maximum uncommitted CPU as all the containers would require. The field is optional with no default value.
+
+
+`limits.memory`::
+The memory limit for the container, corresponding directly to {k8s-resource-request-limit-docs-link}[`spec.containers[\].resources.limits.memory`] setting.
+{ProductPlatformName} will limit the containers to this much memory, potentially terminating their pod if they use more.
+The field is optional with no default value.
+
+`limits.cpu`::
+The CPU limit for the container, corresponding directly to {k8s-resource-request-limit-docs-link}[`spec.containers[\].resources.limits.cpu`] setting.
+{ProductPlatformName} will cap the containers CPU usage to this limit. The field is optional with no default value.
+
+More details about resource limits and requests can be found on {k8s-resource-request-limit-docs-link}[{KubernetesName} website].
```

Comment:
Note for self: Divya to add a link to the OpenShift docs if available or revisit if not available.

Reply:
This might do: https://docs.openshift.com/container-platform/3.9/dev_guide/compute_resources.html#dev-compute-resources